tests/_utils/view.js

```diff
@@ -0,0 +1,836 @@
+ * or object with two fields `view` and `selection` when selection ranges were included in data to parse.
+ */
+export function parse( data, options = { } ) {
+	options.order = options.order || [];
+	const viewParser = new ViewParser();
+	const rangeParser = new RangeParser();
+
+	const view = viewParser.parse( data );
+	const ranges = rangeParser.parse( view, options.order );
+
+	// When ranges are present - return object containing view, and selection.
+	if ( ranges.length ) {
+		const selection = new Selection();
+		selection.setRanges( ranges, !!options.lastRangeBackward );
+
+		return {
+			view: view,
+			selection: selection
+		};
+	}
+
+	return view;
+}
+
+/**
+ * Private helper class used for converting ranges represented as text inside view {@link engine.treeView.Text Text nodes}.
+ *
+ * @private
+ */
+class RangeParser {
+	/**
+	 * Parses the view, and returns ranges represented inside {@link engine.treeView.Text Text nodes}.
+	 * Method will remove all occurrences of `{`, `}`, `[` and `]` from found text nodes. If text node is empty after
+	 * the process - it will be removed too.
+	 *
+	 * @param {engine.treeView.Node} node Starting node.
+	 * @param {Array.<Number>} order Ranges order. Each element should represent desired position of the range after
+	 * sorting. For example: `[2, 3, 1]` means that first range will be placed as second, second as third and third as first.
+	 * @returns {Array.<engine.treeView.Range>} Array with ranges found.
+	 */
+	parse( node, order ) {
+		this._positions = [];
+
+		// Remove all range brackets from view nodes and save their positions.
+		this._getPositions( node );
+
+		// Create ranges using gathered positions.
+		let ranges = this._createRanges();
+
+		// Sort ranges if needed.
+		if ( order.length ) {
+			if ( order.length != ranges.length ) {
+				throw new Error(
+					`Parse error - there are ${ ranges.length} ranges found, but ranges order array contains ${ order.length } elements.`
+				);
+			}
+
+			ranges = this._sortRanges( ranges,  order );
+		}
+
+		return ranges;
+	}
+
+	/**
+	 * Gathers positions of brackets inside view tree starting from provided node. Method will remove all occurrences of
+	 * `{`, `}`, `[` and `]` from found text nodes. If text node is empty after the process - it will be removed
+	 * too.
+	 *
+	 * @private
+	 * @param {engine.treeView.Node} node Staring node.
+	 */
+	_getPositions( node ) {
+		if ( node instanceof ViewDocumentFragment || node instanceof ViewElement ) {
+			// Copy elements into the array, when nodes will be removed from parent node this array will still have all the
+			// items needed for iteration.
+			const children = [ ...node.getChildren() ];
+
+			for ( let child of children ) {
+				this._getPositions( child );
+			}
+		}
+
+		if ( node instanceof ViewText ) {
+			const regexp = new RegExp(
+				`[ ${TEXT_RANGE_START_TOKEN}${TEXT_RANGE_END_TOKEN}\\${ELEMENT_RANGE_END_TOKEN}\\${ELEMENT_RANGE_START_TOKEN} ]`,
```

Comment:
For readability we put spaces inside values in template strings: `'foo ${ value }'`.